### PR TITLE
Module name lookup cleanup

### PR DESCRIPTION
**Pull request by slavapestov**

### CODE REVIEW

**Comment by slavapestov** [comment]
@swift-ci Please smoke test

File: lib/IDE/CodeCompletion.cpp [hunk 1]
```diff
@@ -2977,11 +2975,7 @@ class CompletionLookup final : public swift::VisibleDeclConsumer {
       return;
 
     if (!D->hasInterfaceType())
-      TypeResolver->resolveDeclSignature(D);
-    else if (isa<TypeAliasDecl>(D)) {
-      // A TypeAliasDecl might have type set, but not the underlying type.
-      TypeResolver->resolveDeclSignature(D);
```
Comment (jrose-apple) [inline lib/IDE/CodeCompletion.cpp]: What happened to this bit? Just not necessary for other reasons?

**Comment by jrose-apple** [comment]
I'm pretty sure a Clang module's imported modules are never empty in practice because they all import the stdlib (to not make it weird that they can refer to stdlib types), but using an explicit Optional makes more sense to someone reading the code anyway.

**Comment by slavapestov** [comment]
@jrose-apple I haven't tested my hypothesis but it appears that we only add the stdlib to the imports list in getImportedModules(), and not getImportedModulesForLookup().

File: lib/IDE/CodeCompletion.cpp (hunk 1 above)
Comment (slavapestov) [reply lib/IDE/CodeCompletion.cpp]: Yeah, it's been the case for a while that "has interface type" means "has underlying type". Eventually the two will be separate requests.

**Comment by slavapestov** [comment]
@swift-ci Please smoke test

File: lib/AST/DeclContext.cpp [hunk 2]
```diff
@@ -476,20 +476,15 @@ unsigned DeclContext::getSemanticDepth() const {
...
-        return true;
-  } else if (auto *ED = dyn_cast<ExtensionDecl>(this)) {
-    if (auto *CD = dyn_cast<ClassDecl>(ED->getExtendedNominal()))
-      if (!CD->isGenericContext())
-        return true;
-  }
+  // Members of non-generic classes and class extensions can be found by
+  /// dynamic lookup.
+  if (auto *CD = getSelfClassDecl())
+    return !CD->isGenericContext();
```
Comment (jrose-apple) [inline lib/AST/DeclContext.cpp]: Oops. Generic classes have been able to have `@objc` members for a while. Guess we never updated this (and probably shouldn't at this point).

Comment (slavapestov) [reply lib/AST/DeclContext.cpp]: You're right. I preserved this behavior without even thinking about it.

**Comment by slavapestov** [comment]
@swift-ci Please smoke test

**Comment by slavapestov** [comment]
@swift-ci Please test source compatibility

**Comment by slavapestov** [comment]
@swift-ci Please smoke test

**Comment by slavapestov** [comment]
@swift-ci Please test source compatibility

**Comment by slavapestov** [comment]
@swift-ci Please smoke test

**Comment by slavapestov** [comment]
@swift-ci Please smoke test

**Comment by slavapestov** [comment]
@swift-ci Please smoke test

**Pull request merged**